test/e2e/util_gatewayapi_test.go

```diff
@@ -741,3 +855,83 @@ func assertDNSRecord(t *testing.T, recordName types.NamespacedName) error {
 	})
 	return err
 }
+
+// assertVAP checks if the ValidatingAdmissionPolicy of the given name exists, and returns an error if not.
+func assertVAP(t *testing.T, name string) error {
+	t.Helper()
+	vap := &admissionregistrationv1.ValidatingAdmissionPolicy{}
+	return wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 1*time.Minute, false, func(context context.Context) (bool, error) {
+		if err := kclient.Get(context, types.NamespacedName{Name: name}, vap); err != nil {
+			t.Logf("failed to get vap %q: %v, retrying...", name, err)
+			return false, nil
+		}
+		t.Logf("found vap %q", name)
+		return true, nil
+	})
+}
+
+// scaleDeployment scales the deployment with the given name to the specified number of replicas.
+func scaleDeployment(t *testing.T, namespace, name string, replicas int32) error {
+	t.Helper()
+
+	nsName := types.NamespacedName{Namespace: namespace, Name: name}
+	return wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 30*time.Second, false, func(context context.Context) (bool, error) {
+		depl := &appsv1.Deployment{}
+		if err := kclient.Get(context, nsName, depl); err != nil {
+			t.Logf("failed to get deployment %q: %v, retrying...", nsName, err)
+			return false, nil
+		}
+		if *depl.Spec.Replicas != replicas {
+			depl.Spec.Replicas = &replicas
+			if err := kclient.Update(context, depl); err != nil {
+				t.Logf("failed to update deployment %q: %v, retrying...", nsName, err)
+				return false, nil
+			}
+			t.Logf("scaled deployment %q to %d replica(s)", nsName, replicas)
+		}
+		if depl.Status.AvailableReplicas != replicas {
+			t.Logf("deployment %q expected to have %d available replica(s) but got %d, retrying...", nsName, replicas, depl.Status.AvailableReplicas)
+			return false, nil
+		}
+		t.Logf("deployment %q has %d available replica(s)", nsName, replicas)
+		return true, nil
+	})
+}
+
+// vapManager helps to disable the VAP resource which is managed by CVO.
+type vapManager struct {
+	t    *testing.T
+	name string
+}
+
+// newVAPManager returns a new instance of VAPManager.
+func newVAPManager(t *testing.T, vapName string) *vapManager {
+	return &vapManager{
+		t:    t,
+		name: vapName,
+	}
+}
+
+// disable scales down CVO and removes the VAP resource.
+func (m *vapManager) disable() (error, func()) {
+	if err := scaleDeployment(m.t, cvoNamespace, cvoDeploymentName, 0); err != nil {
+		return fmt.Errorf("failed to scale down cvo: %w", err), func() { /*scale down didn't work, nothing to do*/ }
+	}
+	if err := deleteExistingVAP(m.t, m.name); err != nil {
+		return fmt.Errorf("failed to delete vap %q: %w", m.name, err), func() {
+			if err := scaleDeployment(m.t, cvoNamespace, cvoDeploymentName, 1); err != nil {
+				m.t.Errorf("failed to scale up cvo: %v", err)
```

Comment:
Here it is %v.

Reply:
Right, because here we log the error, we don't wrap it like in [the other case](https://github.com/openshift/cluster-ingress-operator/pull/1192#discussion_r1984125500). Here is a small example to demonstrate the difference: [link](https://go.dev/play/p/p0_vzqWkzmv).